crates/pallets/starknet/src/simulations.rs

```diff
@@ -188,26 +188,36 @@ impl<T: Config> Pallet<T> {
     }
 
     pub fn re_execute_transactions(
-        transactions: Vec<UserOrL1HandlerTransaction>,
+        transactions_before: Vec<UserOrL1HandlerTransaction>,
+        transactions_to_trace: Vec<UserOrL1HandlerTransaction>,
     ) -> Result<Result<Vec<TransactionExecutionInfo>, PlaceHolderErrorTypeForFailedStarknetExecution>, DispatchError>
     {
+        if !transactions_before.is_empty() {
+            let _ = storage::transactional::with_transaction(|| {
+                storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(
+                    Self::re_execute_transactions_inner(transactions_before),
+                ))
+            })
+            .map_err(|_| Error::<T>::FailedToCreateATransactionalStorageExecution)?;
+        }
+
         storage::transactional::with_transaction(|| {
             storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(Self::re_execute_transactions_inner(
-                transactions,
+                transactions_to_trace,
             )))
         })
         .map_err(|_| Error::<T>::FailedToCreateATransactionalStorageExecution)?
     }
```

Comment:
This won't work. Both executions will be done in two different database transactions. It will rollback after `storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok` and revert to the original state.

What you have to do is execute both vec of tx inside the same  `storage::transactional::with_transaction`